packages/vite/src/node/server/transformRequest.ts

```diff
@@ -271,6 +271,23 @@ async function loadAndTransform(
     if (map.mappings && !map.sourcesContent) {
       await injectSourcesContent(map, mod.file, logger)
     }
+    for (
+      let sourcesIndex = 0;
+      sourcesIndex < map.sources.length;
+      ++sourcesIndex
+    ) {
+      const sourcePath = map.sources[sourcesIndex]
+
+      // Rewrite sources to relative paths to give debuggers the chance
+      // to resolve and display them in a meaningful way (rather than
+      // with absolute paths).
+      if (path.isAbsolute(sourcePath) && path.isAbsolute(mod.file)) {
```

Comment:
[Astro and Vitest are failing](https://github.com/vitejs/vite/pull/12079#issuecomment-1443521371) because `sourcePath` can be null (we have checks for this every time we transform the `sources` array
```suggestion
      if (sourcePath && path.isAbsolute(sourcePath) && path.isAbsolute(mod.file)) {
```

Reply:
Fixed, thanks. Note that having `null` in `sources` isn't valid for sourcemaps. Is there some later cleanup pass that removes / rewrites the `null`s?

Reply:
Maybe @sapphi-red has more context here. I can't find a cleanup pass to remove these `null` sources. The libs we use seem to imply that sources can be `null`. Maybe something to work out with these libs too?
- [ampproject/remmaping `SourceMap`](https://github.com/ampproject/remapping/blob/8e3acd04e12381945310772a7bc2384a98cd7fb9/src/source-map.ts#L15)
- [trace-mapping `SourceMapV3`](https://github.com/jridgewell/trace-mapping/blob/23ab4a29244444d825044ff8fb9a8d57f41837fa/src/types.ts#L8)

Reply:
I didn't know about this one. It seems magic-string returns `null` sources sometimes. (I guess when `filename` option is not passed to `new MagicString`.)
https://github.com/ampproject/remapping/commit/d640cc667ccb567ddda8c18b4011e0df36919d5b